### PR TITLE
kubeadm: increase ut for cmd/kubeadm/app/phases/addons/proxy

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Increase coverage cmd/kubeadm/app/phases/addons/proxy

Before:

```
        k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy   coverage: 54.9% of statements
```

After this change:

```
        k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy   coverage: 78.0% of statements
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 1]
```diff
@@ -192,3 +198,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+
+func newMockClientForTest(t *testing.T) *clientsetfake.Clientset {
+	client := clientsetfake.NewSimpleClientset()
+	_, err := client.AppsV1().DaemonSets(metav1.NamespaceSystem).Create(context.TODO(), &apps.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DaemonSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("kube-proxy"),
```
Comment (pacoxu) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:302]:
```suggestion
			Name:      "kube-proxy",
```

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 2]
```diff
@@ -192,3 +198,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
 		})
 	}
 }
+
+func Test_printOrCreateKubeProxyObjects(t *testing.T) {
```
Comment (pacoxu) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:202]:
```suggestion
func TestPrintOrCreateKubeProxyObjects(t *testing.T) {
```

**Comment by pacoxu** [comment]
/lgtm
/assign @neolit123
BTW, this just missed the test freeze of v1.28.

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 3]
```diff
@@ -17,17 +17,22 @@ limitations under the License.
...
+
 	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	clientsetfake "k8s.io/client-go/kubernetes/fake"
 	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
 	core "k8s.io/client-go/testing"
-
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:30]: don't remove this line

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 4]
```diff
@@ -192,3 +197,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+
+func TestPrintOrCreateKubeProxyObjects(t *testing.T) {
+	tests := []struct {
+		name          string
+		printManifest bool
+		wantOut       string
+		wantErr       bool
+	}{
+		{
+			name:          "not print Manifest",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:209]:
```suggestion
			name:          "do not print manifest",
```

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 5]
```diff
@@ -192,3 +197,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+			printManifest: false,
+			wantOut:       "[addons] Applied essential addon: kube-proxy\n",
+			wantErr:       false,
+		},
+		{
+			name:          "print Manifest",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:215]:
```suggestion
			name:          "print manifest",
```

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 6]
```diff
@@ -192,3 +197,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out := &bytes.Buffer{}
+			client := newMockClientForTest(t)
+			cmByte := []byte{'\n', 'f', 'o', 'o', '\n'}
+			dsByte := []byte{'\n', 'b', 'a', 'r', '\n'}
+			if err := printOrCreateKubeProxyObjects(cmByte, dsByte, client, out, tt.printManifest); (err != nil) != tt.wantErr {
+				t.Errorf("printOrCreateKubeProxyObjects() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:283]:
```suggestion
				t.Fatalf("printOrCreateKubeProxyObjects() error = %v, wantErr %v", err, tt.wantErr)
```

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 7]
```diff
@@ -192,3 +197,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+				return
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:284]: if err != nil we should return?

File: cmd/kubeadm/app/phases/addons/proxy/proxy_test.go [hunk 8]
```diff
@@ -192,3 +197,120 @@ func TestDaemonSetsHaveSystemNodeCriticalPriorityClassName(t *testing.T) {
...
+			Labels: map[string]string{
+				"k8s-app": "kube-proxy",
+			},
+		},
+		Spec: apps.DaemonSetSpec{
+			Template: v1.PodTemplateSpec{},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating daemonset: %v", err)
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/addons/proxy/proxy_test.go:312]:
```suggestion
		t.Fatalf("error creating DaemonSet: %v", err)
```

**Review by pacoxu — APPROVED** [review]

**Review by kinzhi — COMMENTED** [review]
/lgtm

**Pull request merged**